### PR TITLE
Update printing for primitives (int, double etc) and vectors/sets

### PR DESCRIPTION
**Pull request by kachayev**
Still hard to tell what to do with:

* maps

* refined of refined (see https://github.com/KitApps/schema-refined/issues/1)

* named schemas, like `Coord`

### CODE REVIEW

**Comment by kachayev** [comment]
Okay, named schemas are now covered with `refined'`, I just need to ensure that `with-meta` doesn't break things as it does here

screen shot 2018-06-23 at 5 16 27 pm

**Comment by kachayev** [comment]
Added additional check if we can safely apply `with-meta` 😐

File: src/schema_refined/core.clj [hunk 1]
```diff
@@ -43,6 +43,30 @@
 (defn predicate? [p]
   (satisfies? Predicate p))
 
+(defn schema->str [schema]
+  {:pre [(schema? schema)]}
+  (let [tag (:schema-refined.core/tag (meta schema))]
```
Comment (gsnewmark) [inline src/schema_refined/core.clj]: Why not use `::tag`?

File: src/schema_refined/core.clj [hunk 2]
```diff
@@ -43,6 +43,30 @@
...
+
+      (and (set? schema)
+           (= 1 (count schema)))
+      (format "#{%s}" (schema->str (first schema)))
+
+      (fn? schema)
+      (schema-utils/fn-name schema)
+
+      :else
+      (with-out-str (pr schema)))))
```
Comment (gsnewmark) [inline src/schema_refined/core.clj]: [`pr-str`](https://clojuredocs.org/clojure.core/pr-str)

File: src/schema_refined/core.clj [hunk 3]
```diff
@@ -134,6 +160,29 @@
   {:pre [(schema? dt)]}
   (RefinedSchema. dt (coerce pred)))
 
+(defmacro refined'
```
Comment (gsnewmark) [inline src/schema_refined/core.clj]: Not sure it's a good name, but can't think of a better one :disappointed:

File: test/schema_refined/core_test.clj [hunk 4]
```diff
@@ -823,3 +823,23 @@
     (ok! CounterWithElse {:num 1})
     (ok! CounterWithElse {:num 2})
     (ok! CounterWithElse {:num 100})))
+
+;;
+;; check tough printing cases
+;;
+
+(defn test-print! [schema pattern]
+  (let [dv (with-out-str (pr schema))]
```
Comment (gsnewmark) [inline test/schema_refined/core_test.clj]: `pr-str`

File: src/schema_refined/core.clj (hunk 1 above)
Comment (kachayev) [reply src/schema_refined/core.clj]: Old habits, easy to mess up when refactoring & splitting code to a different namespaces

File: src/schema_refined/core.clj (hunk 2 above)
Comment (kachayev) [reply src/schema_refined/core.clj]: 👍

**Pull request merged**